src/plugins/mf_priority.cpp

```diff
@@ -323,6 +331,54 @@ static void rec_q_cb (flux_t *h,
 }
 
 
+/*
+ * Unpack a payload from an external bulk update service and place it in the
+ * "projects" vector.
+ */
+static void rec_proj_cb (flux_t *h,
+                         flux_msg_handler_t *mh,
+                         const flux_msg_t *msg,
+                         void *arg)
+{
+    char *project = NULL;
+    json_t *data, *jtemp = NULL;
+    json_error_t error;
+    int num_data = 0;
+
+    if (flux_request_unpack (msg, NULL, "{s:o}", "data", &data) < 0) {
+        flux_log_error (h, "failed to unpack custom_priority.trigger msg");
+        goto error;
+    }
+
+    if (!data || !json_is_array (data)) {
+        flux_log (h, LOG_ERR, "mf_priority: invalid project info payload");
+        goto error;
+    }
+    num_data = json_array_size (data);
+
+    // clear the projects vector
+    projects.clear ();
+
+    for (int i = 0; i < num_data; i++) {
```

Comment:
This looks like a good place for `json_array_foreach`?

Reply:
I think you're right! I converted this for-loop to use `json_array_foreach ()`. Thanks for pointing that out